index.js

```diff
@@ -50,7 +51,13 @@ function Queue (worker, opts) {
   this.pool.on('finish', function finish (output, data, worker, change) {
     var changeNum = change.change
     debug('finish', changeNum)
+    //
+    // When we are dealing with concurrent changes we can hit cases where the
+    // change here is a seq less than the previous. Lets also keep track of the
+    // highest change in order to serve both cases.
+    //
```

Comment:
inline comment++

we should probably write more of these, since this code is pretty dense